desktop-exporter/trace_data_test.go

```diff
@@ -18,22 +18,56 @@ func TestGetTraceSummary(t *testing.T) {
 	store := NewTraceStore(maxQueueLength)
 	spans := extractSpans(ctx, traces)
 
-	// Assign each span start and end time before adding it to the store
-	// These timestamps are used to validate the summary's durationMS
 	for i, span := range spans {
 		span.TraceID = "1"
-		span.StartTime = time.Date(2022, 10, 10, 0, 0, i, 0, time.UTC)
-		span.EndTime = time.Date(2022, 10, 10, 0, 0, i+1, 0, time.UTC)
+		span.SpanID = string(rune(i));
+		// make 0 the root span.
+		if i == 0 {
+			span.Name = "rootSpan"
+			span.ParentSpanID = ""
+			span.Resource.Attributes["service.name"] = "service name"
+		} else {
+			span.ParentSpanID = "0";
+		}
 		store.Add(ctx, span)
 	}
 
 	trace, err := store.GetTrace("1")
 	assert.NoError(t, err)
 
-	summary, err := trace.GetTraceSummary()
+	summary := trace.GetTraceSummary()
+	assert.True(t, summary.HasRootSpan)
+	assert.Equal(t, "service name", summary.RootServiceName)
+	assert.Equal(t, "rootSpan", summary.RootName)
+	assert.Equal(t, time.Date(2022, 10, 21, 7, 10, 2, 100, time.UTC), summary.RootStartTime)
+	assert.Equal(t, time.Date(2020, 10, 21, 7, 10, 2, 300, time.UTC), summary.RootEndTime)
+	assert.Equal(t, uint32(spansPerTrace), summary.SpanCount)
+	assert.Equal(t, "1", summary.TraceID)
+}
+
+func TestGetTraceSummaryMissingRootSpan(t *testing.T) {
+	maxQueueLength := 1
+	spansPerTrace := 3
+
+	traces := testdata.GenerateOTLPPayload(1, 1, maxQueueLength*spansPerTrace)
```

Comment:
I didn't realize `GenerateOTLPPayload` doesn't create any root spans. It would be good to add that as a comment

```suggestion
	// All spans from testdata.GenerateOTLPPayload have the same parent span ID
	traces := testdata.GenerateOTLPPayload(1, 1, maxQueueLength*spansPerTrace)
```